vinyldns.go

```diff
@@ -142,6 +142,58 @@ func main() {
 				},
 			},
 		},
+		{
+			Name:        "zone-create",
+			Usage:       "zone-create --name <name> --email <email> --admin-group-id <adminGroupID> --transfer-connection-name <transferConnectionName> --transfer-connection-key <transferConnectionKey> --transfer-connection-key-name <transferConnectionKeyName> --transfer-connection-primary-server <transferConnectionPrimaryServer> --zone-connection-name <zoneConnectionName> --zone-connection-key <zoneConnectionKey> --zone-connection-key-name <zoneConnectionKeyName> --zone-connection-primary-server <zoneConnectionPrimaryServer>",
+			Description: "Create a zone",
+			Action:      zoneCreate,
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "name",
+					Usage: "The zone name",
+				},
+				cli.StringFlag{
+					Name:  "email",
+					Usage: "The zone email",
+				},
+				cli.StringFlag{
+					Name:  "admin-group-id",
+					Usage: "The zone admin group ID",
+				},
+				cli.StringFlag{
+					Name:  "transfer-connection-name",
```

Comment:
I know technically we have this `connection name` thing, but it is not used in practice for anything.  I wonder to simplify things slightly if we omit it and just force it to be the same value as the key name.

Reply:
Are you saying that `Name` and `KeyName` are the same thing and therefore redundant on transfer and zone connections?

Reply:
`Name` is just not used, meaningless from an app standpoint.  Really should be deprecated / deleted in a future version of vinyldns.

Reply:
Awesome; I'll remove it.

Reply:
@pauljamescleary Maybe I'm a bit confused. When I remove `Connection.Name`, I get the following -- did I misinterpret you?

```
 ✗ zone-create
   (in test file tests/vinyldns_tests.bats, line 23)
     `)' failed
   Request URL:
   http://localhost:9000/zones
   Request Method:
   POST
   Request body:
   {"name":"ok.","email":"test@test.com","adminGroupId":"ok-group","connection":{"keyName":"vinyldns.","key":"nzisn+4G2ldMn0q1CV3vsg==","primaryServer":"vinyldns-bind9"},"transferConnection":{"keyName":"vinyldns.","key":"nzisn+4G2ldMn0q1CV3vsg==","primaryServer":"vinyldns-bind9"}}
   Response code:
   400
   Response body:
   {"errors":["Missing ZoneConnection.name"]}
```

Reply:
Right, if you _do_ put the connection you have to put the connection name in.  Sorry if that wasn't clear.

My comment was, from a cli standpoint, you don't need to ask the user for it.  In your code, just set the connection name to the same value as the keyName.

Reply:
Ah, gotcha. Thanks for clarifying!